app/commands/v2/unpublish.rb

```diff
@@ -17,7 +22,7 @@ def call
 
         check_version_and_raise_if_conflicting(content_item, previous_version_number)
 
-        if draft_present?(content_id)
+        if draft_present?(content_id) && !payload[:allow_draft]
```

Comment:
I wonder if there's a cleaner way to express this. Perhaps extracting a method and adding `!payload[:allow_draft]` as a guard clause?

It's a shame that the `discard_drafts` behaviour is intertwined with `allow_draft`. I was hoping the "cannot be used together" error would flatten out some of this logic but I guess that's not the case?